src/incremental/compareCIL.ml

```diff
@@ -192,28 +192,32 @@ let eqF_check_contained_renames ~renameDetection f1 f2 oldMap newMap cfgs gc_old
      otherwise check that the function comparison was successful without collecting any rename assumptions *)
   let dependenciesMatch, change_info, final_matches =
     if renameDetection then
-      let funDependenciesMatch, change_info, final_matches =
-        let extract_globs _ gc map =
-          let v = get_varinfo gc in
-          VarinfoMap.add v gc map in
-        let var_glob_old = GlobalMap.fold extract_globs oldMap VarinfoMap.empty in
-        let var_glob_new = GlobalMap.fold extract_globs newMap VarinfoMap.empty in
-        VarinfoMap.fold (fun f_old_var f_new_var (acc, ci, fm) ->
-            match VarinfoMap.find_opt f_old_var (fst fm) with
-            | None ->
-              let f_old = get_fundec (VarinfoMap.find f_old_var var_glob_old) in
-              let f_new = get_fundec (VarinfoMap.find f_new_var var_glob_new) in (* TODO: what happens if there exists no fundec for this varinfo? *)
-              if acc then
-                eqF_only_consider_exact_match f_old f_new ci fm oldMap newMap var_glob_old var_glob_new
-              else false, ci, fm
-            | Some v -> v.vid = f_new_var.vid, ci, fm) function_dependencies (true, change_info, final_matches) in
+      let extract_globs _ gc map =
+        let v = get_varinfo gc in
+        VarinfoMap.add v gc map in
+      let var_glob_old = GlobalMap.fold extract_globs oldMap VarinfoMap.empty in
+      let var_glob_new = GlobalMap.fold extract_globs newMap VarinfoMap.empty in
+      let funDependenciesMatch, change_info, final_matches = VarinfoMap.fold (fun f_old_var f_new_var (acc, ci, fm) ->
+          let glob_old = VarinfoMap.find f_old_var var_glob_old in
+          let glob_new = VarinfoMap.find f_new_var var_glob_new in
+          match VarinfoMap.find_opt f_old_var (fst fm) with
+          | None ->
+            let f_old = get_fundec glob_old in
+            let f_new = get_fundec glob_new in (* TODO: what happens if there exists no fundec for this varinfo? *)
+            if acc then
+              eqF_only_consider_exact_match f_old f_new ci fm oldMap newMap var_glob_old var_glob_new
+            else false, ci, fm
+          | Some v -> acc && v.vid = f_new_var.vid && List.mem {old=glob_old; current=glob_new} ci.unchanged, ci, fm
```

Comment:
It seems to me that these `global_col` things may contain `varinfo`s and `fundec`s, which shouldn't be polymorphically compared like `List.mem` does.

`BatList.mem_cmp` or `List.exists` on only the IDs or something would be better.